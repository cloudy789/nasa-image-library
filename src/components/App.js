import React, { Component } from 'react';
import '../styles/css/App.css';

import SearchBar from './SearchBar';
import Gallery from './Gallery';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { searchData: {} };

    this.handleSearch = this.handleSearch.bind(this);
  }

  componentDidMount() {
    this.handleSearch({
      searchTerm: 'galaxy',
      image: true,
      video: false,
      audio: false,
    });
  }

  async handleSearch(searchObj) {
    const mediaTypes = [];
    let query = '';
    Object.entries(searchObj).forEach(([key, value]) => {
      if (key !== 'searchTerm' && value === true) {
        mediaTypes.push(key);
      } else if (value !== false) {
        query = value;
      }
    });
    const url = `https://images-api.nasa.gov/search?q=${query}&media_type=${mediaTypes.toString()}`;
    const response = await fetch(url);
    const data = await response.json();
    this.setState({
      searchData: data.collection,
    });
  }

  render() {
    return (
      <div className="wrapper">
        <div className="app-header">
          <h2>Nasa Image Library</h2>
          <SearchBar onSearch={this.handleSearch} />
        </div>
        <Gallery galleryData={this.state.searchData} />
      </div>
    );
  }
}

export default App;
