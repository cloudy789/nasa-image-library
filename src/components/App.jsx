import React, { useState } from 'react';
import styled from 'styled-components/macro';

import useFetch from '../hooks/useFetch';
import Header from './header/Header';
import Gallery from './gallery/Gallery';
import Loading from './shared/Loading';
import Error from './shared/Error';
import GalleryModal from './modal/GalleryModal';
import GalleryModalContent from './modal/GalleryModalContent';

const StyledApp = styled.div`
  width: 100%;
  min-width: 200px;
  margin: 0 auto;
  padding: 0;
  box-sizing: border-box;
  background: ${({ theme }) => theme.ghostWhite};
  height: 100vh;
  ${({ clickedModalMetadata }) => clickedModalMetadata && 'overflow: hidden;'}
`;

const App = () => {
  const [clickedModalMetadata, setClickedModalMetadata] = useState(null);
  const [{ fetchedData, isLoading, isError }, doFetch] = useFetch(
    'https://images-api.nasa.gov/search?q=iss&media_type=image',
    {
      collection: { items: [] },
    },
  );

  const { collection: data } = fetchedData;

  return (
    <StyledApp clickedModalMetadata={clickedModalMetadata}>
      <Header doFetch={doFetch} />
      {isError && <Error />}
      {isLoading ? (
        <Loading />
      ) : (
        <Gallery
          setClickedModalMetadata={setClickedModalMetadata}
          data={data}
          doFetch={doFetch}
        />
      )}
      {clickedModalMetadata && (
        <GalleryModal setClickedModalMetadata={setClickedModalMetadata}>
          <GalleryModalContent clickedModalMetadata={clickedModalMetadata} />
        </GalleryModal>
      )}
    </StyledApp>
  );
};

export default App;
