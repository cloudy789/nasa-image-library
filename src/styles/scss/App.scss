@import './variablesMixins.scss';

.fetch-errors-modal {
    text-align: center;
}
