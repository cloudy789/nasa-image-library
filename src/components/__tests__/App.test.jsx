import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

test('renders without throwing errors', async () => {
  const container = render(<App />);
  expect(container).toMatchSnapshot();
});
