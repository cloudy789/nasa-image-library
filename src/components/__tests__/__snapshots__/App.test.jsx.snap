// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without throwing errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App__StyledApp-oine8m-0 gGdzqt"
      >
        <div
          class="Header__StyledHeader-sc-19c87cm-0 gRkUaE"
        >
          <a
            class="Header___StyledLink-sc-19c87cm-1 jdgJVX"
            href="/"
          >
            <h1>
              NASA 
              <span>
                Media Library
              </span>
            </h1>
          </a>
          <div
            class="SearchBar__StyledSearchBar-f1a6f2-0 dJDgne"
          >
            <form>
              <div
                class="SearchBarInput__StyledSearchBarInput-sc-17b1y68-0 URyXi"
              >
                <input
                  name="searchTerm"
                  placeholder="e.g. ISS"
                  type="text"
                  value=""
                />
                <button
                  type="submit"
                >
                  <span
                    aria-hidden="true"
                    class="fa fa-search"
                  />
                </button>
              </div>
              <label
                class="SearchBarCheckbox__StyledSearchBarCheckbox-vs5p6j-0 gslsDk"
              >
                Images
                <input
                  checked=""
                  name="searchImages"
                  type="checkbox"
                />
              </label>
              <label
                class="SearchBarCheckbox__StyledSearchBarCheckbox-vs5p6j-0 gslsDk"
              >
                Videos
                <input
                  name="searchVideos"
                  type="checkbox"
                />
              </label>
              <label
                class="SearchBarCheckbox__StyledSearchBarCheckbox-vs5p6j-0 gslsDk"
              >
                Audio
                <input
                  name="searchAudio"
                  type="checkbox"
                />
              </label>
            </form>
          </div>
        </div>
        <main
          class="App__StyledMain-oine8m-1 frUcWr"
        >
          <div
            class="Loading__StyledLoading-sc-119z7o2-0 kwZJmS"
          >
            <i
              class="fa fa-circle-o-notch fa-spin"
            />
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App__StyledApp-oine8m-0 gGdzqt"
    >
      <div
        class="Header__StyledHeader-sc-19c87cm-0 gRkUaE"
      >
        <a
          class="Header___StyledLink-sc-19c87cm-1 jdgJVX"
          href="/"
        >
          <h1>
            NASA 
            <span>
              Media Library
            </span>
          </h1>
        </a>
        <div
          class="SearchBar__StyledSearchBar-f1a6f2-0 dJDgne"
        >
          <form>
            <div
              class="SearchBarInput__StyledSearchBarInput-sc-17b1y68-0 URyXi"
            >
              <input
                name="searchTerm"
                placeholder="e.g. ISS"
                type="text"
                value=""
              />
              <button
                type="submit"
              >
                <span
                  aria-hidden="true"
                  class="fa fa-search"
                />
              </button>
            </div>
            <label
              class="SearchBarCheckbox__StyledSearchBarCheckbox-vs5p6j-0 gslsDk"
            >
              Images
              <input
                checked=""
                name="searchImages"
                type="checkbox"
              />
            </label>
            <label
              class="SearchBarCheckbox__StyledSearchBarCheckbox-vs5p6j-0 gslsDk"
            >
              Videos
              <input
                name="searchVideos"
                type="checkbox"
              />
            </label>
            <label
              class="SearchBarCheckbox__StyledSearchBarCheckbox-vs5p6j-0 gslsDk"
            >
              Audio
              <input
                name="searchAudio"
                type="checkbox"
              />
            </label>
          </form>
        </div>
      </div>
      <main
        class="App__StyledMain-oine8m-1 frUcWr"
      >
        <div
          class="Loading__StyledLoading-sc-119z7o2-0 kwZJmS"
        >
          <i
            class="fa fa-circle-o-notch fa-spin"
          />
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
